flax/linen/module.py

```diff
@@ -49,6 +49,45 @@ def _check_omnistaging():
         "  config.enable_omnistaging()")
 
 
+def _indent(x: str, num_spaces: int):
+  indent_str = ' ' * num_spaces
+  lines = x.split('\n')
+  # skip last line because it is always empty and should not be indented.
```

Comment:
Isn't this just because of the way we construct the string in module_repr()? I think it would be better to not make this assumption here and handle both cases.

Reply:
Done